tmt/utils.py

```diff
@@ -2632,21 +2634,40 @@ def remove_color(text: str) -> str:
     return re.sub(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])', '', text)
 
 
-def default_branch(repository: Path, remote: str = 'origin') -> Optional[str]:
+def default_branch(repository: Path, logger: tmt.log.Logger,
+                   remote: str = 'origin') -> Optional[str]:
     """ Detect default branch from given local git repository """
-    # Make sure the '.git/refs/remotes/{remote}' directory is present
-    git_remotes_dir = repository / f'.git/refs/remotes/{remote}'
-    if not git_remotes_dir.exists():
+    # Make sure '.git' (which is a file or a directory) is present
+    dot_git = repository / '.git'
+    if not dot_git.exists():
+        return None
+
+    if not dot_git.is_file() and not dot_git.is_dir():
         return None
 
-    head = repository / f'.git/refs/remotes/{remote}/HEAD'
-    # Make sure the HEAD reference is available
-    if not head.exists():
-        subprocess.run(
-            f'git remote set-head {remote} --auto'.split(), cwd=repository)
-    # The ref format is 'ref: refs/remotes/origin/main'
-    with open(head) as ref:
-        return ref.read().strip().split('/')[-1]
+    if dot_git.is_dir():
+        # Make sure the '.git/refs/remotes/{remote}' directory is present
+        git_remotes_dir = repository / f'.git/refs/remotes/{remote}'
+        if not git_remotes_dir.exists():
+            return None
+
+        head = repository / f'.git/refs/remotes/{remote}/HEAD'
+        # Make sure the HEAD reference is available
+        if not head.exists():
+            subprocess.run(
+                f'git remote set-head {remote} --auto'.split(), cwd=repository)
+        # The ref format is 'ref: refs/remotes/origin/main'
+        with open(head) as ref:
+            return ref.read().strip().split('/')[-1]
+
+    # Get the default branch if '.git' is a file
+    worktree_path = dot_git.read_text()
+    if '/.git/' not in worktree_path:
+        return None
+    git_parent_path = re.sub(r'/?\.git/.*', '', worktree_path)
```

Comment:
Thinking about it a bit more: Instead of parsing this special `.git` file, would it make sense to use this?

    git rev-parse --path-format=absolute --git-common-dir

Seems to be more straightforward, and also probably a bit safe not to depend on internal format of that file.

Reply:
Updated according to your suggestion.

Reply:
This suggestion was meant as a better alternative for line 2667 above.

Reply:
> This suggestion was meant as a better alternative for line 2667 above.

After talking to openAI, I finally got what you mean. (sorry that my head was a mess after having 10 days for PTO).

What you mean `git rev-parse --path-format=absolute --git-common-dir` is we should use it to get the parent path if `.git` is a file. Sorry again!